aws/lambda/clickhouse-replicator-dynamo/lambda_function.py

```diff
@@ -0,0 +1,195 @@
+import json
+import os
+import re
+from collections import defaultdict
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+from warnings import warn
+
+import clickhouse_connect
+
+
+DYNAMODB_TABLE_REGEX = re.compile(
+    "arn:aws:dynamodb:.*?:.*?:table/(?P<table>[0-9a-zA-Z_-]+)/.+"
+)
+CLICKHOUSE_ENDPOINT = os.getenv("CLICKHOUSE_ENDPOINT", "")
+CLICKHOUSE_USERNAME = os.getenv("CLICKHOUSE_USERNAME", "default")
+CLICKHOUSE_PASSWORD = os.getenv("CLICKHOUSE_PASSWORD", "")
+SUPPORTED_TABLES = {
+    "torchci-workflow-job": "workflow_job",
+    "torchci-workflow-run": "workflow_run",
+    "torchci-push": "push",
+    "torchci-pull-request": "pull_request",
+    "torchci-issues": "issues",
+}
+
+class EventType(Enum):
+    INSERT = "INSERT"
+    REMOVE = "REMOVE"
+    MODIFY = "MODIFY"
+
+
+def lambda_handler(event: Any, context: Any) -> None:
+    # https://clickhouse.com/docs/en/integrations/python
+    clickhouse_client = clickhouse_connect.get_client(
```

Comment:
I think it will help with the lambda perf if we move this `get_client` call outside of the lambda_handler function to the global scope.  This is to reduce the cold start of running the lambda https://docs.aws.amazon.com/lambda/latest/operatorguide/execution-environments.html#cold-starts

Reply:
I think you mean warm start time?

I moved it to a function with a lru cache but I don't know if this helps with the start up time

I have mild worries that the client would die if kept alive too long?